dm/syncer/error.go

```diff
@@ -61,6 +64,93 @@ func isDropColumnWithIndexError(err error) bool {
 			strings.Contains(mysqlErr.Message, "with tidb_enable_change_multi_schema is disable"))
 }
 
+// here db should be TiDB database.
+func GetDDLStatusFromTiDB(db *sql.DB, ddl string, createTime int64) (string, error) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	rowNum := 10
+	count := 0
+	for {
+		showJobs := fmt.Sprintf("ADMIN SHOW DDL JOBS %d", rowNum)
+		// every attempt try 10 history jobs
+		rows, err := db.QueryContext(ctx, showJobs)
+		if err != nil {
+			return "", err
+		}
+		defer rows.Close()
+
+		columns, err := rows.Columns()
+		if err != nil {
+			return "", err
+		}
+
+		values := make([]sql.RawBytes, len(columns))
+		scanArgs := make([]interface{}, len(values))
+		for i := range values {
+			scanArgs[i] = &values[i]
+		}
+
+		// skip the lines that have been checked
+		skipNum := 0
+		if rowNum > 10 {
+			for rows.Next() {
+				skipNum++
+				if skipNum >= rowNum-10 {
+					break
+				}
+			}
+		}
+		for rows.Next() {
+			err = rows.Scan(scanArgs...)
+			if err != nil {
+				return "", err
+			}
+
+			createTimeStr := string(values[8])
+			timeLayout := "2006-01-02 15:04:05"
+			loc, _ := time.LoadLocation("Local")
+			theTime, _ := time.ParseInLocation(timeLayout, createTimeStr, loc)
```

Comment:
check if this value is related to downstream timezone, and left an comment

Reply:
Modified